signalfx/ingest.py

```diff
@@ -271,31 +273,55 @@ def _set_event_properties(self, pbuf_obj, properties):
             prop.key = key
             self._assign_property_value(prop, value)
 
-    def _assign_property_value(self, prop, value):
-        if isinstance(value, int):
-            prop.value.intValue = value
-        elif isinstance(value, str):
-            prop.value.strValue = value
-        elif isinstance(value, unicode):
-            prop.value.strValue = value
-        elif isinstance(value, float):
-            prop.value.doubleValue = value
-        elif isinstance(value, bool):
-            prop.value.boolValue = value
-        else:
-            raise ValueError('Invalid Value ' + str(value))
-
-    def _assign_value_type(self, pbuf_dp, value):
-        if isinstance(value, int):
-            pbuf_dp.value.intValue = value
-        elif isinstance(value, str):
-            pbuf_dp.value.strValue = value
-        elif isinstance(value, unicode):
-            pbuf_dp.value.strValue = value
+    def _assign_value_by_type(self, pbuf_obj, value, _bool=True, _float=True,
+                        _integer=True, _string=True):
+        """Assigns the supplied value to the appropriate protobuf value type"""
+        # bool inherits int, so bool instance check must be executed prior to
+        # checking for integer types
+        if isinstance(value, bool):
+            if _bool is True:
```

Comment:
Addressed your concerns and added type of invalid variable to the raised ValueError.